src/context/archiveContext.js

```diff
@@ -0,0 +1,87 @@
+import { createContext, useContext } from "react"
+import axios from "axios";
+import { useNotes } from "./notesContext";
+import { useTrash } from "./trashContext";
+
+const ArchiveContext = createContext();
+
+function ArchiveProvider({ children }) {
+
+    const encodedToken = localStorage.getItem("token");
+    const { noteDispatch } = useNotes();
+    const { trashHandler } = useTrash()
+
+    const archiveHandler = async (note) => {
+        const { _id } = note;
+        try {
+            const response = await axios.post(
+                `/api/notes/archives/${_id}`,
+                {
+                    note: note,
+                },
+                {
+                    headers: {
+                        authorization: encodedToken,
+                    },
+                }
+            );
+            if (response.status === 201) {
+                noteDispatch({ type: "SET_ARCHIVE", payload: response.data.archives })
+                noteDispatch({ type: "SET_NOTES", payload: response.data.notes })
```

Comment:
You can use a single dispatch by adding another state in the notes reducer called "MOVE_TO_ARCHIVE" and update both archived and notes in that state.